src/mlpack/methods/ann/layer_names.hpp

```diff
@@ -0,0 +1,316 @@
+/**
+ * @file layer_names.hpp
+ * @author Sreenik Seal
+ *
+ * Implementation of a class that converts a given ann layer to string format.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <boost/variant/static_visitor.hpp>
+#include <string>
+
+using namespace mlpack::ann;
+
+/**
+ * Implementation of a class that returns the string representation of the
+ * name of the given layer.
+ */
+class LayerNameVisitor : public boost::static_visitor<std::string>
+{
+ public:
+  //! Create the LayerNameVisitor object.
+  LayerNameVisitor()
+  {
+  }
+
+  /*
+   * Return the name of the given layer of type AtrousConvolution as a string.
+   * 
+   * @param Given layer of type AtrousConvolution.
+   * @return The string representation of the layer.
+   */
+  std::string LayerString(AtrousConvolution<>* /*layer*/) const
+  {
+    return "atrousconvolution";
+  }
+
+  /*
+   * Return the name of the given layer of type AlphaDropout as a string.
+   * 
+   * @param Given layer of type AlphaDropout.
+   * @return The string representation of the layer.
+   */
+  std::string LayerString(AlphaDropout<>* /*layer*/) const
+  {
+    return "alphadropout";
+  }
```

Comment:
Do you think it would be easier if, e.g., we held a `const static std::string layerName` in each layer type and then for this visitor, we simply detected using SFINAE whether that layer type had a `layerName` member, and returned that?  (or `unsupported` if it doesn't exist.)  I know that this style was @zoq's suggestion, so maybe it is better that way, but I wanted to toss this possibility out there. :)